updated the documentation, reflecting the changes in #306

lduer opened the pull request:
## Description
Missing documentation updates in the developers-section about the `kimai.invoice.documens` and removed parts for `renderer`, `calculator` and `number_generator`

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [ ] My code follows the code style
- [ ] All files have a license header 
- [ ] All methods have a doc header with type declarations
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes

kevinpapst commented:
Ups, seems as if I missed something ;-) Thanks for the notification!
Whileyour changes are true for the invoice renderer, the documentation for the calculator and number-generator is only partially wrong. They should not be deleted but updated. Could you add something like this:
```
## Adding invoice calculator

An invoice calculator is a class extending `App\Invoice\CalculatorInterface` and is responsible for calculating 
invoice rates, taxes and taking care of all timesheet entries that should be displayed.   

Every invoice calculator class will be picked up when refreshing the application cache by the [InvoiceServiceCompilerPass](https://github.com/kevinpapst/kimai2/blob/master/src/DependencyInjection/Compiler/InvoiceServiceCompilerPass.php):

The ID of the calculator must be unique, please prefix it with your vendor or bundle name and make sure it only contains
character as it will be stored in a database column.

## Adding invoice-number generator

An invoice-number generator is a class extending `App\Invoice\NumberGeneratorInterface` and its only task is to generate 
a number for the invoice. In most cases you do not want to mix multiple invoice-number generators throughout your invoices.   

Every invoice number-generator class will be picked up when refreshing the application cache by the [InvoiceServiceCompilerPass](https://github.com/kevinpapst/kimai2/blob/master/src/DependencyInjection/Compiler/InvoiceServiceCompilerPass.php):

The ID of the number generator must be unique, please prefix it with your vendor or bundle name and make sure it only contains
character as it will be stored in a database column.
```
Then I will merge the changes.

lduer commented:
I added your text to the developer docs and added some words to the translation docs about the new translation-files for the invoice-caclulator, the invoice number-generator and the invoice-documents.

kevinpapst commented:
Merci Lukas! Really appreciate your contribution 👍
